java.lang.ClassCastException: org.apache.shardingsphere.sql.parser.sql.segment.dml.column.ColumnSegment cannot be cast to org.apache.shardingsphere.sql.parser.sql.segment.dml.expr.ExpressionSegment

shardingsphere 4.0.1 is ok, but 4.1.0 throws an exception？

shardingsphere : 4.1.0

SQL:  

```sql
update ic_goods_occupy_record set actual_amount = pre_amount, pre_amount = 0, version = version + 1, update_time = now() ,update_person = ? where biz_bill_no = ? or orig_bill_no = ? and is_deleted = 0
```

```
org.mybatis.spring.MyBatisSystemException: nested exception is 
org.apache.ibatis.exceptions.PersistenceException: ### Error updating database. Cause: java.lang.ClassCastException: 
org.apache.shardingsphere.sql.parser.sql.segment.dml.column.ColumnSegment cannot be cast to org.apache.shardingsphere.sql.parser.sql.segment.dml.expr.ExpressionSegment 
### T The error may involve defaultParameterMap 
### The error occurred while setting parameters 
### SQL: update ic_goods_occupy_record set actual_amount = pre_amount, pre_amount = 0, version = version + 1, update_time = now() ,update_person = ? where biz_bill_no = ? or orig_bill_no = ? and is_deleted = 0 
### Cause: java.lang.ClassCastException: org.apache.shardingsphere.sql.parser.sql.segment.dml.column.ColumnSegment cannot be cast to org.apache.shardingsphere.sql.parser.sql.segment.dml.expr.ExpressionSegment 
at org.mybatis.spring.MyBatisExceptionTranslator.translateExceptionIfPossible(MyBatisExceptionTranslator.java:78) 
at org.mybatis.spring.SqlSessionTemplate$SqlSessionInterceptor.invoke(SqlSessionTemplate.java:440) 
at com.sun.proxy.$Proxy73.update(Unknown Source) at org.mybatis.spring.SqlSessionTemplate.update(SqlSessionTemplate.java:287) at org.apache.ibatis.binding.MapperMethod.execute(MapperMethod.java:67) at org.apache.ibatis.binding.MapperProxy.invoke(MapperProxy.java:57) at com.sun.proxy.$Proxy107.updateActualAmountByOrderNo(Unknown Source) 
at com.ncarzone.saleinventory.biz.component.goodinv.GoodsOccupyRecordComponent.updateActualAmountByOrderNo(GoodsOccupyRecordComponent.java:60)
at com.ncarzone.saleinventory.biz.component.goodinv.GoodsOccupyRecordComponent$$FastClassBySpringCGLIB$$fc1b0c75.invoke() 
at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204) at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:684) at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204) at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:746) at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:294) at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:98) at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:185) at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:88) sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:644) at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:633) at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:70) at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:174) at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:88) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:644) at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:633) at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:70) at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:174) at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:92) at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:185) at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:688)
```

I got it.

```
[ERROR] 14:36:51.737 [ShardingSphere-Command-1] o.a.s.s.f.c.CommandExecutorTask - Exception occur:
java.lang.ClassCastException: org.apache.shardingsphere.sql.parser.sql.segment.dml.column.ColumnSegment cannot be cast to org.apache.shardingsphere.sql.parser.sql.segment.dml.expr.ExpressionSegment
        at org.apache.shardingsphere.sql.parser.mysql.visitor.impl.MySQLDMLVisitor.visitAssignment(MySQLDMLVisitor.java:242)
        at org.apache.shardingsphere.sql.parser.mysql.visitor.impl.MySQLDMLVisitor.visitAssignment(MySQLDMLVisitor.java:127)
        at org.apache.shardingsphere.sql.parser.autogen.MySQLStatementParser$AssignmentContext.accept(MySQLStatementParser.java:1753)
        at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visit(AbstractParseTreeVisitor.java:18)
        at org.apache.shardingsphere.sql.parser.mysql.visitor.impl.MySQLDMLVisitor.visitSetAssignmentsClause(MySQLDMLVisitor.java:225)
        at org.apache.shardingsphere.sql.parser.mysql.visitor.impl.MySQLDMLVisitor.visitSetAssignmentsClause(MySQLDMLVisitor.java:127)
        at org.apache.shardingsphere.sql.parser.autogen.MySQLStatementParser$SetAssignmentsClauseContext.accept(MySQLStatementParser.java:1801)
        at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visit(AbstractParseTreeVisitor.java:18)
        at org.apache.shardingsphere.sql.parser.mysql.visitor.impl.MySQLDMLVisitor.visitUpdate(MySQLDMLVisitor.java:213)
        at org.apache.shardingsphere.sql.parser.mysql.visitor.impl.MySQLDMLVisitor.visitUpdate(MySQLDMLVisitor.java:127)
        at org.apache.shardingsphere.sql.parser.autogen.MySQLStatementParser$UpdateContext.accept(MySQLStatementParser.java:1622)
        at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visit(AbstractParseTreeVisitor.java:18)
        at org.apache.shardingsphere.sql.parser.SQLParserEngine.parse0(SQLParserEngine.java:80)
        at org.apache.shardingsphere.sql.parser.SQLParserEngine.parse(SQLParserEngine.java:61)
        at org.apache.shardingsphere.shardingproxy.backend.text.TextProtocolBackendHandlerFactory.newInstance(TextProtocolBackendHandlerFactory.java:69)
        at org.apache.shardingsphere.shardingproxy.frontend.mysql.command.query.text.query.MySQLComQueryPacketExecutor.<init>(MySQLComQueryPacketExecutor.java:65)
        at org.apache.shardingsphere.shardingproxy.frontend.mysql.command.MySQLCommandExecutorFactory.newInstance(MySQLCommandExecutorFactory.java:70)
        at org.apache.shardingsphere.shardingproxy.frontend.mysql.command.MySQLCommandExecuteEngine.getCommandExecutor(MySQLCommandExecuteEngine.java:59)
        at org.apache.shardingsphere.shardingproxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:92)
        at org.apache.shardingsphere.shardingproxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:71)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)

```

--- a/sharding-integration-test/sharding-jdbc-test/src/test/resources/integrate/cases/dml/dataset/db/update_with_column_equal_column.xml
+++ b/sharding-integration-test/sharding-jdbc-test/src/test/resources/integrate/cases/dml/dataset/db/update_with_column_equal_column.xml
@@ -1,0 +1,64 @@
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one or more
+  ~ contributor license agreements.  See the NOTICE file distributed with
+  ~ this work for additional information regarding copyright ownership.
+  ~ The ASF licenses this file to You under the Apache License, Version 2.0
+  ~ (the "License"); you may not use this file except in compliance with
+  ~ the License.  You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<dataset update-count="1">
+    <metadata data-nodes="db_${0..9}.t_order">
+        <column name="order_id" type="numeric" />
+        <column name="user_id" type="numeric" />
+        <column name="status" type="varchar" />
+    </metadata>
+    <row data-node="db_0.t_order" values="1000, 10, init" />
+    <row data-node="db_0.t_order" values="1001, 10, init" />
+    <row data-node="db_0.t_order" values="2000, 20, init" />
+    <row data-node="db_0.t_order" values="2001, 20, init" />
+    <row data-node="db_1.t_order" values="1100, 11, init" />
+    <row data-node="db_1.t_order" values="1101, 11, init" />
+    <row data-node="db_1.t_order" values="2100, 21, init" />
+    <row data-node="db_1.t_order" values="2101, 21, init" />
+    <row data-node="db_2.t_order" values="1200, 12, init" />
+    <row data-node="db_2.t_order" values="1201, 12, init" />
+    <row data-node="db_2.t_order" values="2200, 22, init" />
+    <row data-node="db_2.t_order" values="2201, 22, init" />
+    <row data-node="db_3.t_order" values="1300, 13, init" />
+    <row data-node="db_3.t_order" values="1301, 13, init" />
+    <row data-node="db_3.t_order" values="2300, 23, init" />
+    <row data-node="db_3.t_order" values="2301, 23, init" />
+    <row data-node="db_4.t_order" values="1400, 14, init" />
+    <row data-node="db_4.t_order" values="1401, 14, init" />
+    <row data-node="db_4.t_order" values="2400, 24, init" />
+    <row data-node="db_4.t_order" values="2401, 24, init" />
+    <row data-node="db_5.t_order" values="1500, 15, init" />
+    <row data-node="db_5.t_order" values="1501, 15, init" />
+    <row data-node="db_5.t_order" values="2500, 25, init" />
+    <row data-node="db_5.t_order" values="2501, 25, init" />
+    <row data-node="db_6.t_order" values="1600, 16, init" />
+    <row data-node="db_6.t_order" values="1601, 16, init" />
+    <row data-node="db_6.t_order" values="2600, 26, init" />
+    <row data-node="db_6.t_order" values="2601, 26, init" />
+    <row data-node="db_7.t_order" values="1700, 17, init" />
+    <row data-node="db_7.t_order" values="1701, 17, init" />
+    <row data-node="db_7.t_order" values="2700, 27, init" />
+    <row data-node="db_7.t_order" values="2701, 27, init" />
+    <row data-node="db_8.t_order" values="1800, 18, init" />
+    <row data-node="db_8.t_order" values="1801, 18, init" />
+    <row data-node="db_8.t_order" values="2800, 28, init" />
+    <row data-node="db_8.t_order" values="2801, 28, init" />
+    <row data-node="db_9.t_order" values="1900, 19, init" />
+    <row data-node="db_9.t_order" values="1901, 19, init" />
+    <row data-node="db_9.t_order" values="2900, 29, init" />
+    <row data-node="db_9.t_order" values="2901, 29, init" />
+</dataset>

--- a/sharding-integration-test/sharding-jdbc-test/src/test/resources/integrate/cases/dml/dataset/dbtbl_with_masterslave/update_with_column_equal_column.xml
+++ b/sharding-integration-test/sharding-jdbc-test/src/test/resources/integrate/cases/dml/dataset/dbtbl_with_masterslave/update_with_column_equal_column.xml
@@ -1,0 +1,224 @@
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one or more
+  ~ contributor license agreements.  See the NOTICE file distributed with
+  ~ this work for additional information regarding copyright ownership.
+  ~ The ASF licenses this file to You under the Apache License, Version 2.0
+  ~ (the "License"); you may not use this file except in compliance with
+  ~ the License.  You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<dataset update-count="1">
+    <metadata data-nodes="master_db_${0..9}.t_order_${0..9},slave_db_${0..9}.t_order_${0..9}">
+        <column name="order_id" type="numeric" />
+        <column name="user_id" type="numeric" />
+        <column name="status" type="varchar" />
+    </metadata>
+    <row data-node="master_db_0.t_order_0" values="1000, 10, init" />
+    <row data-node="master_db_0.t_order_1" values="1001, 10, init" />
+    <row data-node="master_db_0.t_order_2" values="1002, 10, init" />
+    <row data-node="master_db_0.t_order_3" values="1003, 10, init" />
+    <row data-node="master_db_0.t_order_4" values="1004, 10, init" />
+    <row data-node="master_db_0.t_order_5" values="1005, 10, init" />
+    <row data-node="master_db_0.t_order_6" values="1006, 10, init" />
+    <row data-node="master_db_0.t_order_7" values="1007, 10, init" />
+    <row data-node="master_db_0.t_order_8" values="1008, 10, init" />
+    <row data-node="master_db_0.t_order_9" values="1009, 10, init" />
+    <row data-node="master_db_1.t_order_0" values="1100, 11, init" />
+    <row data-node="master_db_1.t_order_1" values="1101, 11, init" />
+    <row data-node="master_db_1.t_order_2" values="1102, 11, init" />
+    <row data-node="master_db_1.t_order_3" values="1103, 11, init" />
+    <row data-node="master_db_1.t_order_4" values="1104, 11, init" />
+    <row data-node="master_db_1.t_order_5" values="1105, 11, init" />
+    <row data-node="master_db_1.t_order_6" values="1106, 11, init" />
+    <row data-node="master_db_1.t_order_7" values="1107, 11, init" />
+    <row data-node="master_db_1.t_order_8" values="1108, 11, init" />
+    <row data-node="master_db_1.t_order_9" values="1109, 11, init" />
+    <row data-node="master_db_2.t_order_0" values="1200, 12, init" />
+    <row data-node="master_db_2.t_order_1" values="1201, 12, init" />
+    <row data-node="master_db_2.t_order_2" values="1202, 12, init" />
+    <row data-node="master_db_2.t_order_3" values="1203, 12, init" />
+    <row data-node="master_db_2.t_order_4" values="1204, 12, init" />
+    <row data-node="master_db_2.t_order_5" values="1205, 12, init" />
+    <row data-node="master_db_2.t_order_6" values="1206, 12, init" />
+    <row data-node="master_db_2.t_order_7" values="1207, 12, init" />
+    <row data-node="master_db_2.t_order_8" values="1208, 12, init" />
+    <row data-node="master_db_2.t_order_9" values="1209, 12, init" />
+    <row data-node="master_db_3.t_order_0" values="1300, 13, init" />
+    <row data-node="master_db_3.t_order_1" values="1301, 13, init" />
+    <row data-node="master_db_3.t_order_2" values="1302, 13, init" />
+    <row data-node="master_db_3.t_order_3" values="1303, 13, init" />
+    <row data-node="master_db_3.t_order_4" values="1304, 13, init" />
+    <row data-node="master_db_3.t_order_5" values="1305, 13, init" />
+    <row data-node="master_db_3.t_order_6" values="1306, 13, init" />
+    <row data-node="master_db_3.t_order_7" values="1307, 13, init" />
+    <row data-node="master_db_3.t_order_8" values="1308, 13, init" />
+    <row data-node="master_db_3.t_order_9" values="1309, 13, init" />
+    <row data-node="master_db_4.t_order_0" values="1400, 14, init" />
+    <row data-node="master_db_4.t_order_1" values="1401, 14, init" />
+    <row data-node="master_db_4.t_order_2" values="1402, 14, init" />
+    <row data-node="master_db_4.t_order_3" values="1403, 14, init" />
+    <row data-node="master_db_4.t_order_4" values="1404, 14, init" />
+    <row data-node="master_db_4.t_order_5" values="1405, 14, init" />
+    <row data-node="master_db_4.t_order_6" values="1406, 14, init" />
+    <row data-node="master_db_4.t_order_7" values="1407, 14, init" />
+    <row data-node="master_db_4.t_order_8" values="1408, 14, init" />
+    <row data-node="master_db_4.t_order_9" values="1409, 14, init" />
+    <row data-node="master_db_5.t_order_0" values="1500, 15, init" />
+    <row data-node="master_db_5.t_order_1" values="1501, 15, init" />
+    <row data-node="master_db_5.t_order_2" values="1502, 15, init" />
+    <row data-node="master_db_5.t_order_3" values="1503, 15, init" />
+    <row data-node="master_db_5.t_order_4" values="1504, 15, init" />
+    <row data-node="master_db_5.t_order_5" values="1505, 15, init" />
+    <row data-node="master_db_5.t_order_6" values="1506, 15, init" />
+    <row data-node="master_db_5.t_order_7" values="1507, 15, init" />
+    <row data-node="master_db_5.t_order_8" values="1508, 15, init" />
+    <row data-node="master_db_5.t_order_9" values="1509, 15, init" />
+    <row data-node="master_db_6.t_order_0" values="1600, 16, init" />
+    <row data-node="master_db_6.t_order_1" values="1601, 16, init" />
+    <row data-node="master_db_6.t_order_2" values="1602, 16, init" />
+    <row data-node="master_db_6.t_order_3" values="1603, 16, init" />
+    <row data-node="master_db_6.t_order_4" values="1604, 16, init" />
+    <row data-node="master_db_6.t_order_5" values="1605, 16, init" />
+    <row data-node="master_db_6.t_order_6" values="1606, 16, init" />
+    <row data-node="master_db_6.t_order_7" values="1607, 16, init" />
+    <row data-node="master_db_6.t_order_8" values="1608, 16, init" />
+    <row data-node="master_db_6.t_order_9" values="1609, 16, init" />
+    <row data-node="master_db_7.t_order_0" values="1700, 17, init" />
+    <row data-node="master_db_7.t_order_1" values="1701, 17, init" />
+    <row data-node="master_db_7.t_order_2" values="1702, 17, init" />
+    <row data-node="master_db_7.t_order_3" values="1703, 17, init" />
+    <row data-node="master_db_7.t_order_4" values="1704, 17, init" />
+    <row data-node="master_db_7.t_order_5" values="1705, 17, init" />
+    <row data-node="master_db_7.t_order_6" values="1706, 17, init" />
+    <row data-node="master_db_7.t_order_7" values="1707, 17, init" />
+    <row data-node="master_db_7.t_order_8" values="1708, 17, init" />
+    <row data-node="master_db_7.t_order_9" values="1709, 17, init" />
+    <row data-node="master_db_8.t_order_0" values="1800, 18, init" />
+    <row data-node="master_db_8.t_order_1" values="1801, 18, init" />
+    <row data-node="master_db_8.t_order_2" values="1802, 18, init" />
+    <row data-node="master_db_8.t_order_3" values="1803, 18, init" />
+    <row data-node="master_db_8.t_order_4" values="1804, 18, init" />
+    <row data-node="master_db_8.t_order_5" values="1805, 18, init" />
+    <row data-node="master_db_8.t_order_6" values="1806, 18, init" />
+    <row data-node="master_db_8.t_order_7" values="1807, 18, init" />
+    <row data-node="master_db_8.t_order_8" values="1808, 18, init" />
+    <row data-node="master_db_8.t_order_9" values="1809, 18, init" />
+    <row data-node="master_db_9.t_order_0" values="1900, 19, init" />
+    <row data-node="master_db_9.t_order_1" values="1901, 19, init" />
+    <row data-node="master_db_9.t_order_2" values="1902, 19, init" />
+    <row data-node="master_db_9.t_order_3" values="1903, 19, init" />
+    <row data-node="master_db_9.t_order_4" values="1904, 19, init" />
+    <row data-node="master_db_9.t_order_5" values="1905, 19, init" />
+    <row data-node="master_db_9.t_order_6" values="1906, 19, init" />
+    <row data-node="master_db_9.t_order_7" values="1907, 19, init" />
+    <row data-node="master_db_9.t_order_8" values="1908, 19, init" />
+    <row data-node="master_db_9.t_order_9" values="1909, 19, init" />
+    <row data-node="slave_db_0.t_order_0" values="1000, 10, init_slave" />
+    <row data-node="slave_db_0.t_order_1" values="1001, 10, init_slave" />
+    <row data-node="slave_db_0.t_order_2" values="1002, 10, init_slave" />
+    <row data-node="slave_db_0.t_order_3" values="1003, 10, init_slave" />
+    <row data-node="slave_db_0.t_order_4" values="1004, 10, init_slave" />
+    <row data-node="slave_db_0.t_order_5" values="1005, 10, init_slave" />
+    <row data-node="slave_db_0.t_order_6" values="1006, 10, init_slave" />
+    <row data-node="slave_db_0.t_order_7" values="1007, 10, init_slave" />
+    <row data-node="slave_db_0.t_order_8" values="1008, 10, init_slave" />
+    <row data-node="slave_db_0.t_order_9" values="1009, 10, init_slave" />
+    <row data-node="slave_db_1.t_order_0" values="1100, 11, init_slave" />
+    <row data-node="slave_db_1.t_order_1" values="1101, 11, init_slave" />
+    <row data-node="slave_db_1.t_order_2" values="1102, 11, init_slave" />
+    <row data-node="slave_db_1.t_order_3" values="1103, 11, init_slave" />
+    <row data-node="slave_db_1.t_order_4" values="1104, 11, init_slave" />
+    <row data-node="slave_db_1.t_order_5" values="1105, 11, init_slave" />
+    <row data-node="slave_db_1.t_order_6" values="1106, 11, init_slave" />
+    <row data-node="slave_db_1.t_order_7" values="1107, 11, init_slave" />
+    <row data-node="slave_db_1.t_order_8" values="1108, 11, init_slave" />
+    <row data-node="slave_db_1.t_order_9" values="1109, 11, init_slave" />
+    <row data-node="slave_db_2.t_order_0" values="1200, 12, init_slave" />
+    <row data-node="slave_db_2.t_order_1" values="1201, 12, init_slave" />
+    <row data-node="slave_db_2.t_order_2" values="1202, 12, init_slave" />
+    <row data-node="slave_db_2.t_order_3" values="1203, 12, init_slave" />
+    <row data-node="slave_db_2.t_order_4" values="1204, 12, init_slave" />
+    <row data-node="slave_db_2.t_order_5" values="1205, 12, init_slave" />
+    <row data-node="slave_db_2.t_order_6" values="1206, 12, init_slave" />
+    <row data-node="slave_db_2.t_order_7" values="1207, 12, init_slave" />
+    <row data-node="slave_db_2.t_order_8" values="1208, 12, init_slave" />
+    <row data-node="slave_db_2.t_order_9" values="1209, 12, init_slave" />
+    <row data-node="slave_db_3.t_order_0" values="1300, 13, init_slave" />
+    <row data-node="slave_db_3.t_order_1" values="1301, 13, init_slave" />
+    <row data-node="slave_db_3.t_order_2" values="1302, 13, init_slave" />
+    <row data-node="slave_db_3.t_order_3" values="1303, 13, init_slave" />
+    <row data-node="slave_db_3.t_order_4" values="1304, 13, init_slave" />
+    <row data-node="slave_db_3.t_order_5" values="1305, 13, init_slave" />
+    <row data-node="slave_db_3.t_order_6" values="1306, 13, init_slave" />
+    <row data-node="slave_db_3.t_order_7" values="1307, 13, init_slave" />
+    <row data-node="slave_db_3.t_order_8" values="1308, 13, init_slave" />
+    <row data-node="slave_db_3.t_order_9" values="1309, 13, init_slave" />
+    <row data-node="slave_db_4.t_order_0" values="1400, 14, init_slave" />
+    <row data-node="slave_db_4.t_order_1" values="1401, 14, init_slave" />
+    <row data-node="slave_db_4.t_order_2" values="1402, 14, init_slave" />
+    <row data-node="slave_db_4.t_order_3" values="1403, 14, init_slave" />
+    <row data-node="slave_db_4.t_order_4" values="1404, 14, init_slave" />
+    <row data-node="slave_db_4.t_order_5" values="1405, 14, init_slave" />
+    <row data-node="slave_db_4.t_order_6" values="1406, 14, init_slave" />
+    <row data-node="slave_db_4.t_order_7" values="1407, 14, init_slave" />
+    <row data-node="slave_db_4.t_order_8" values="1408, 14, init_slave" />
+    <row data-node="slave_db_4.t_order_9" values="1409, 14, init_slave" />
+    <row data-node="slave_db_5.t_order_0" values="1500, 15, init_slave" />
+    <row data-node="slave_db_5.t_order_1" values="1501, 15, init_slave" />
+    <row data-node="slave_db_5.t_order_2" values="1502, 15, init_slave" />
+    <row data-node="slave_db_5.t_order_3" values="1503, 15, init_slave" />
+    <row data-node="slave_db_5.t_order_4" values="1504, 15, init_slave" />
+    <row data-node="slave_db_5.t_order_5" values="1505, 15, init_slave" />
+    <row data-node="slave_db_5.t_order_6" values="1506, 15, init_slave" />
+    <row data-node="slave_db_5.t_order_7" values="1507, 15, init_slave" />
+    <row data-node="slave_db_5.t_order_8" values="1508, 15, init_slave" />
+    <row data-node="slave_db_5.t_order_9" values="1509, 15, init_slave" />
+    <row data-node="slave_db_6.t_order_0" values="1600, 16, init_slave" />
+    <row data-node="slave_db_6.t_order_1" values="1601, 16, init_slave" />
+    <row data-node="slave_db_6.t_order_2" values="1602, 16, init_slave" />
+    <row data-node="slave_db_6.t_order_3" values="1603, 16, init_slave" />
+    <row data-node="slave_db_6.t_order_4" values="1604, 16, init_slave" />
+    <row data-node="slave_db_6.t_order_5" values="1605, 16, init_slave" />
+    <row data-node="slave_db_6.t_order_6" values="1606, 16, init_slave" />
+    <row data-node="slave_db_6.t_order_7" values="1607, 16, init_slave" />
+    <row data-node="slave_db_6.t_order_8" values="1608, 16, init_slave" />
+    <row data-node="slave_db_6.t_order_9" values="1609, 16, init_slave" />
+    <row data-node="slave_db_7.t_order_0" values="1700, 17, init_slave" />
+    <row data-node="slave_db_7.t_order_1" values="1701, 17, init_slave" />
+    <row data-node="slave_db_7.t_order_2" values="1702, 17, init_slave" />
+    <row data-node="slave_db_7.t_order_3" values="1703, 17, init_slave" />
+    <row data-node="slave_db_7.t_order_4" values="1704, 17, init_slave" />
+    <row data-node="slave_db_7.t_order_5" values="1705, 17, init_slave" />
+    <row data-node="slave_db_7.t_order_6" values="1706, 17, init_slave" />
+    <row data-node="slave_db_7.t_order_7" values="1707, 17, init_slave" />
+    <row data-node="slave_db_7.t_order_8" values="1708, 17, init_slave" />
+    <row data-node="slave_db_7.t_order_9" values="1709, 17, init_slave" />
+    <row data-node="slave_db_8.t_order_0" values="1800, 18, init_slave" />
+    <row data-node="slave_db_8.t_order_1" values="1801, 18, init_slave" />
+    <row data-node="slave_db_8.t_order_2" values="1802, 18, init_slave" />
+    <row data-node="slave_db_8.t_order_3" values="1803, 18, init_slave" />
+    <row data-node="slave_db_8.t_order_4" values="1804, 18, init_slave" />
+    <row data-node="slave_db_8.t_order_5" values="1805, 18, init_slave" />
+    <row data-node="slave_db_8.t_order_6" values="1806, 18, init_slave" />
+    <row data-node="slave_db_8.t_order_7" values="1807, 18, init_slave" />
+    <row data-node="slave_db_8.t_order_8" values="1808, 18, init_slave" />
+    <row data-node="slave_db_8.t_order_9" values="1809, 18, init_slave" />
+    <row data-node="slave_db_9.t_order_0" values="1900, 19, init_slave" />
+    <row data-node="slave_db_9.t_order_1" values="1901, 19, init_slave" />
+    <row data-node="slave_db_9.t_order_2" values="1902, 19, init_slave" />
+    <row data-node="slave_db_9.t_order_3" values="1903, 19, init_slave" />
+    <row data-node="slave_db_9.t_order_4" values="1904, 19, init_slave" />
+    <row data-node="slave_db_9.t_order_5" values="1905, 19, init_slave" />
+    <row data-node="slave_db_9.t_order_6" values="1906, 19, init_slave" />
+    <row data-node="slave_db_9.t_order_7" values="1907, 19, init_slave" />
+    <row data-node="slave_db_9.t_order_8" values="1908, 19, init_slave" />
+    <row data-node="slave_db_9.t_order_9" values="1909, 19, init_slave" />
+</dataset>

--- a/sharding-integration-test/sharding-jdbc-test/src/test/resources/integrate/cases/dml/dataset/masterslave/update_with_column_equal_column.xml
+++ b/sharding-integration-test/sharding-jdbc-test/src/test/resources/integrate/cases/dml/dataset/masterslave/update_with_column_equal_column.xml
@@ -1,0 +1,144 @@
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one or more
+  ~ contributor license agreements.  See the NOTICE file distributed with
+  ~ this work for additional information regarding copyright ownership.
+  ~ The ASF licenses this file to You under the Apache License, Version 2.0
+  ~ (the "License"); you may not use this file except in compliance with
+  ~ the License.  You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<dataset update-count="1">
+    <metadata data-nodes="master.t_order,slave_${0..1}.t_order">
+        <column name="order_id" type="numeric" />
+        <column name="user_id" type="numeric" />
+        <column name="status" type="varchar" />
+    </metadata>
+    <row data-node="master.t_order" values="1000, 10, init" />
+    <row data-node="master.t_order" values="1001, 10, init" />
+    <row data-node="master.t_order" values="1100, 11, init" />
+    <row data-node="master.t_order" values="1101, 11, init" />
+    <row data-node="master.t_order" values="1200, 12, init" />
+    <row data-node="master.t_order" values="1201, 12, init" />
+    <row data-node="master.t_order" values="1300, 13, init" />
+    <row data-node="master.t_order" values="1301, 13, init" />
+    <row data-node="master.t_order" values="1400, 14, init" />
+    <row data-node="master.t_order" values="1401, 14, init" />
+    <row data-node="master.t_order" values="1500, 15, init" />
+    <row data-node="master.t_order" values="1501, 15, init" />
+    <row data-node="master.t_order" values="1600, 16, init" />
+    <row data-node="master.t_order" values="1601, 16, init" />
+    <row data-node="master.t_order" values="1700, 17, init" />
+    <row data-node="master.t_order" values="1701, 17, init" />
+    <row data-node="master.t_order" values="1800, 18, init" />
+    <row data-node="master.t_order" values="1801, 18, init" />
+    <row data-node="master.t_order" values="1900, 19, init" />
+    <row data-node="master.t_order" values="1901, 19, init" />
+    <row data-node="master.t_order" values="2000, 20, init" />
+    <row data-node="master.t_order" values="2001, 20, init" />
+    <row data-node="master.t_order" values="2100, 21, init" />
+    <row data-node="master.t_order" values="2101, 21, init" />
+    <row data-node="master.t_order" values="2200, 22, init" />
+    <row data-node="master.t_order" values="2201, 22, init" />
+    <row data-node="master.t_order" values="2300, 23, init" />
+    <row data-node="master.t_order" values="2301, 23, init" />
+    <row data-node="master.t_order" values="2400, 24, init" />
+    <row data-node="master.t_order" values="2401, 24, init" />
+    <row data-node="master.t_order" values="2500, 25, init" />
+    <row data-node="master.t_order" values="2501, 25, init" />
+    <row data-node="master.t_order" values="2600, 26, init" />
+    <row data-node="master.t_order" values="2601, 26, init" />
+    <row data-node="master.t_order" values="2700, 27, init" />
+    <row data-node="master.t_order" values="2701, 27, init" />
+    <row data-node="master.t_order" values="2800, 28, init" />
+    <row data-node="master.t_order" values="2801, 28, init" />
+    <row data-node="master.t_order" values="2900, 29, init" />
+    <row data-node="master.t_order" values="2901, 29, init" />
+    <row data-node="slave_0.t_order" values="1000, 10, init_slave" />
+    <row data-node="slave_0.t_order" values="1001, 10, init_slave" />
+    <row data-node="slave_0.t_order" values="1100, 11, init_slave" />
+    <row data-node="slave_0.t_order" values="1101, 11, init_slave" />
+    <row data-node="slave_0.t_order" values="1200, 12, init_slave" />
+    <row data-node="slave_0.t_order" values="1201, 12, init_slave" />
+    <row data-node="slave_0.t_order" values="1300, 13, init_slave" />
+    <row data-node="slave_0.t_order" values="1301, 13, init_slave" />
+    <row data-node="slave_0.t_order" values="1400, 14, init_slave" />
+    <row data-node="slave_0.t_order" values="1401, 14, init_slave" />
+    <row data-node="slave_0.t_order" values="1500, 15, init_slave" />
+    <row data-node="slave_0.t_order" values="1501, 15, init_slave" />
+    <row data-node="slave_0.t_order" values="1600, 16, init_slave" />
+    <row data-node="slave_0.t_order" values="1601, 16, init_slave" />
+    <row data-node="slave_0.t_order" values="1700, 17, init_slave" />
+    <row data-node="slave_0.t_order" values="1701, 17, init_slave" />
+    <row data-node="slave_0.t_order" values="1800, 18, init_slave" />
+    <row data-node="slave_0.t_order" values="1801, 18, init_slave" />
+    <row data-node="slave_0.t_order" values="1900, 19, init_slave" />
+    <row data-node="slave_0.t_order" values="1901, 19, init_slave" />
+    <row data-node="slave_0.t_order" values="2000, 20, init_slave" />
+    <row data-node="slave_0.t_order" values="2001, 20, init_slave" />
+    <row data-node="slave_0.t_order" values="2100, 21, init_slave" />
+    <row data-node="slave_0.t_order" values="2101, 21, init_slave" />
+    <row data-node="slave_0.t_order" values="2200, 22, init_slave" />
+    <row data-node="slave_0.t_order" values="2201, 22, init_slave" />
+    <row data-node="slave_0.t_order" values="2300, 23, init_slave" />
+    <row data-node="slave_0.t_order" values="2301, 23, init_slave" />
+    <row data-node="slave_0.t_order" values="2400, 24, init_slave" />
+    <row data-node="slave_0.t_order" values="2401, 24, init_slave" />
+    <row data-node="slave_0.t_order" values="2500, 25, init_slave" />
+    <row data-node="slave_0.t_order" values="2501, 25, init_slave" />
+    <row data-node="slave_0.t_order" values="2600, 26, init_slave" />
+    <row data-node="slave_0.t_order" values="2601, 26, init_slave" />
+    <row data-node="slave_0.t_order" values="2700, 27, init_slave" />
+    <row data-node="slave_0.t_order" values="2701, 27, init_slave" />
+    <row data-node="slave_0.t_order" values="2800, 28, init_slave" />
+    <row data-node="slave_0.t_order" values="2801, 28, init_slave" />
+    <row data-node="slave_0.t_order" values="2900, 29, init_slave" />
+    <row data-node="slave_0.t_order" values="2901, 29, init_slave" />
+    <row data-node="slave_1.t_order" values="1000, 10, init_slave" />
+    <row data-node="slave_1.t_order" values="1001, 10, init_slave" />
+    <row data-node="slave_1.t_order" values="1100, 11, init_slave" />
+    <row data-node="slave_1.t_order" values="1101, 11, init_slave" />
+    <row data-node="slave_1.t_order" values="1200, 12, init_slave" />
+    <row data-node="slave_1.t_order" values="1201, 12, init_slave" />
+    <row data-node="slave_1.t_order" values="1300, 13, init_slave" />
+    <row data-node="slave_1.t_order" values="1301, 13, init_slave" />
+    <row data-node="slave_1.t_order" values="1400, 14, init_slave" />
+    <row data-node="slave_1.t_order" values="1401, 14, init_slave" />
+    <row data-node="slave_1.t_order" values="1500, 15, init_slave" />
+    <row data-node="slave_1.t_order" values="1501, 15, init_slave" />
+    <row data-node="slave_1.t_order" values="1600, 16, init_slave" />
+    <row data-node="slave_1.t_order" values="1601, 16, init_slave" />
+    <row data-node="slave_1.t_order" values="1700, 17, init_slave" />
+    <row data-node="slave_1.t_order" values="1701, 17, init_slave" />
+    <row data-node="slave_1.t_order" values="1800, 18, init_slave" />
+    <row data-node="slave_1.t_order" values="1801, 18, init_slave" />
+    <row data-node="slave_1.t_order" values="1900, 19, init_slave" />
+    <row data-node="slave_1.t_order" values="1901, 19, init_slave" />
+    <row data-node="slave_1.t_order" values="2000, 20, init_slave" />
+    <row data-node="slave_1.t_order" values="2001, 20, init_slave" />
+    <row data-node="slave_1.t_order" values="2100, 21, init_slave" />
+    <row data-node="slave_1.t_order" values="2101, 21, init_slave" />
+    <row data-node="slave_1.t_order" values="2200, 22, init_slave" />
+    <row data-node="slave_1.t_order" values="2201, 22, init_slave" />
+    <row data-node="slave_1.t_order" values="2300, 23, init_slave" />
+    <row data-node="slave_1.t_order" values="2301, 23, init_slave" />
+    <row data-node="slave_1.t_order" values="2400, 24, init_slave" />
+    <row data-node="slave_1.t_order" values="2401, 24, init_slave" />
+    <row data-node="slave_1.t_order" values="2500, 25, init_slave" />
+    <row data-node="slave_1.t_order" values="2501, 25, init_slave" />
+    <row data-node="slave_1.t_order" values="2600, 26, init_slave" />
+    <row data-node="slave_1.t_order" values="2601, 26, init_slave" />
+    <row data-node="slave_1.t_order" values="2700, 27, init_slave" />
+    <row data-node="slave_1.t_order" values="2701, 27, init_slave" />
+    <row data-node="slave_1.t_order" values="2800, 28, init_slave" />
+    <row data-node="slave_1.t_order" values="2801, 28, init_slave" />
+    <row data-node="slave_1.t_order" values="2900, 29, init_slave" />
+    <row data-node="slave_1.t_order" values="2901, 29, init_slave" />
+</dataset>

--- a/sharding-integration-test/sharding-jdbc-test/src/test/resources/integrate/cases/dml/dataset/shadow/update_with_column_equal_column.xml
+++ b/sharding-integration-test/sharding-jdbc-test/src/test/resources/integrate/cases/dml/dataset/shadow/update_with_column_equal_column.xml
@@ -1,0 +1,64 @@
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one or more
+  ~ contributor license agreements.  See the NOTICE file distributed with
+  ~ this work for additional information regarding copyright ownership.
+  ~ The ASF licenses this file to You under the Apache License, Version 2.0
+  ~ (the "License"); you may not use this file except in compliance with
+  ~ the License.  You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<dataset update-count="1">
+    <metadata data-nodes="db.t_order">
+        <column name="order_id" type="numeric" />
+        <column name="user_id" type="numeric" />
+        <column name="status" type="varchar" />
+    </metadata>
+    <row data-node="db.t_order" values="1000, 10, init" />
+    <row data-node="db.t_order" values="1001, 10, init" />
+    <row data-node="db.t_order" values="1100, 11, init" />
+    <row data-node="db.t_order" values="1101, 11, init" />
+    <row data-node="db.t_order" values="1200, 12, init" />
+    <row data-node="db.t_order" values="1201, 12, init" />
+    <row data-node="db.t_order" values="1300, 13, init" />
+    <row data-node="db.t_order" values="1301, 13, init" />
+    <row data-node="db.t_order" values="1400, 14, init" />
+    <row data-node="db.t_order" values="1401, 14, init" />
+    <row data-node="db.t_order" values="1500, 15, init" />
+    <row data-node="db.t_order" values="1501, 15, init" />
+    <row data-node="db.t_order" values="1600, 16, init" />
+    <row data-node="db.t_order" values="1601, 16, init" />
+    <row data-node="db.t_order" values="1700, 17, init" />
+    <row data-node="db.t_order" values="1701, 17, init" />
+    <row data-node="db.t_order" values="1800, 18, init" />
+    <row data-node="db.t_order" values="1801, 18, init" />
+    <row data-node="db.t_order" values="1900, 19, init" />
+    <row data-node="db.t_order" values="1901, 19, init" />
+    <row data-node="db.t_order" values="2000, 20, init" />
+    <row data-node="db.t_order" values="2001, 20, init" />
+    <row data-node="db.t_order" values="2100, 21, init" />
+    <row data-node="db.t_order" values="2101, 21, init" />
+    <row data-node="db.t_order" values="2200, 22, init" />
+    <row data-node="db.t_order" values="2201, 22, init" />
+    <row data-node="db.t_order" values="2300, 23, init" />
+    <row data-node="db.t_order" values="2301, 23, init" />
+    <row data-node="db.t_order" values="2400, 24, init" />
+    <row data-node="db.t_order" values="2401, 24, init" />
+    <row data-node="db.t_order" values="2500, 25, init" />
+    <row data-node="db.t_order" values="2501, 25, init" />
+    <row data-node="db.t_order" values="2600, 26, init" />
+    <row data-node="db.t_order" values="2601, 26, init" />
+    <row data-node="db.t_order" values="2700, 27, init" />
+    <row data-node="db.t_order" values="2701, 27, init" />
+    <row data-node="db.t_order" values="2800, 28, init" />
+    <row data-node="db.t_order" values="2801, 28, init" />
+    <row data-node="db.t_order" values="2900, 29, init" />
+    <row data-node="db.t_order" values="2901, 29, init" />
+</dataset>

--- a/sharding-integration-test/sharding-jdbc-test/src/test/resources/integrate/cases/dml/dataset/tbl/update_with_column_equal_column.xml
+++ b/sharding-integration-test/sharding-jdbc-test/src/test/resources/integrate/cases/dml/dataset/tbl/update_with_column_equal_column.xml
@@ -1,0 +1,44 @@
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one or more
+  ~ contributor license agreements.  See the NOTICE file distributed with
+  ~ this work for additional information regarding copyright ownership.
+  ~ The ASF licenses this file to You under the Apache License, Version 2.0
+  ~ (the "License"); you may not use this file except in compliance with
+  ~ the License.  You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<dataset update-count="1">
+    <metadata data-nodes="tbl.t_order_${0..9}">
+        <column name="order_id" type="numeric" />
+        <column name="user_id" type="numeric" />
+        <column name="status" type="varchar" />
+    </metadata>
+    <row data-node="tbl.t_order_0" values="1000, 10, init" />
+    <row data-node="tbl.t_order_1" values="1001, 10, init" />
+    <row data-node="tbl.t_order_2" values="1002, 10, init" />
+    <row data-node="tbl.t_order_3" values="1003, 10, init" />
+    <row data-node="tbl.t_order_4" values="1004, 10, init" />
+    <row data-node="tbl.t_order_5" values="1005, 10, init" />
+    <row data-node="tbl.t_order_6" values="1006, 10, init" />
+    <row data-node="tbl.t_order_7" values="1007, 10, init" />
+    <row data-node="tbl.t_order_8" values="1008, 10, init" />
+    <row data-node="tbl.t_order_9" values="1009, 10, init" />
+    <row data-node="tbl.t_order_0" values="1100, 11, init" />
+    <row data-node="tbl.t_order_1" values="1101, 11, init" />
+    <row data-node="tbl.t_order_2" values="1102, 11, init" />
+    <row data-node="tbl.t_order_3" values="1103, 11, init" />
+    <row data-node="tbl.t_order_4" values="1104, 11, init" />
+    <row data-node="tbl.t_order_5" values="1105, 11, init" />
+    <row data-node="tbl.t_order_6" values="1106, 11, init" />
+    <row data-node="tbl.t_order_7" values="1107, 11, init" />
+    <row data-node="tbl.t_order_8" values="1108, 11, init" />
+    <row data-node="tbl.t_order_9" values="1109, 11, init" />
+</dataset>

--- a/sharding-integration-test/sharding-jdbc-test/src/test/resources/integrate/cases/dml/dml-integrate-test-cases.xml
+++ b/sharding-integration-test/sharding-jdbc-test/src/test/resources/integrate/cases/dml/dml-integrate-test-cases.xml
@@ -157,6 +157,10 @@
         <assertion parameters="1000:int, 0:int, 10:int" expected-data-file="update.xml" />
     </dml-test-case>
 
+    <dml-test-case sql-case-id="update_with_column_equal_column">
+        <assertion expected-data-file="update_with_column_equal_column.xml" />
+    </dml-test-case>
+
     <dml-test-case sql-case-id="delete_with_sharding_value">
         <assertion parameters="1000:int, 10:int, init:String" expected-data-file="delete_with_sharding_value.xml" />
     </dml-test-case>

--- a/sharding-sql-test/src/main/resources/sql/supported/dml/update.xml
+++ b/sharding-sql-test/src/main/resources/sql/supported/dml/update.xml
@@ -26,4 +26,5 @@
     <sql-case id="update_with_special_comments" value="UPDATE /*+ index(field1) */ ONLY t_order SET status=? WHERE order_id = ? AND user_id = ? RETURN * LOG ERRORS INTO TABLE_LOG" db-types="Oracle" />
     <sql-case id="update_without_parameters" value="UPDATE t_order SET status = 'update' WHERE order_id = 1000 AND user_id = 10" />
     <sql-case id="update_with_or" value="UPDATE t_order SET status = 'update' WHERE (order_id = ? OR order_id = ?) AND user_id = ?" />
+    <sql-case id="update_with_column_equal_column" value="update t_order set status = status where order_id = order_id AND order_id = 1000" db-types="MySQL"/>
 </sql-cases>

--- a/shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/impl/MySQLDMLVisitor.java
+++ b/shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/impl/MySQLDMLVisitor.java
@@ -247,7 +247,12 @@ public final class MySQLDMLVisitor extends MySQLVisitor implements DMLVisitor {
     public ASTNode visitAssignmentValue(final AssignmentValueContext ctx) {
         ExprContext expr = ctx.expr();
         if (null != expr) {
-            return visit(expr);
+            ASTNode result = visit(expr);
+            if (result instanceof ColumnSegment) {
+                return new CommonExpressionSegment(ctx.getStart().getStartIndex(), ctx.getStop().getStopIndex(), ctx.getText());
+            } else {
+                return result;
+            }
         }
         return new CommonExpressionSegment(ctx.getStart().getStartIndex(), ctx.getStop().getStopIndex(), ctx.getText());
     }

--- a/shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/test/resources/sql/dml/update.xml
+++ b/shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/test/resources/sql/dml/update.xml
@@ -347,4 +347,32 @@
             </and-predicate>
         </where>
     </update>
+
+    <update sql-case-id="update_with_column_equal_column" >
+        <table name="t_order" start-index="7" stop-index="13" />
+        <set start-index="15" stop-index="33" literal-stop-index="33">
+            <assignment start-index="19" stop-index="33" literal-stop-index="33">
+                <column name="status" start-index="19" stop-index="24" />
+                <assignment-value>
+                    <common-expression text="status" literal-text="status" start-index="28" stop-index="33" literal-start-index="28" literal-stop-index="33"/>
+                </assignment-value>
+            </assignment>
+        </set>
+        <where start-index="35" stop-index="79" literal-start-index="35" literal-stop-index="79">
+            <and-predicate>
+                <predicate start-index="41" stop-index="59" literal-start-index="41" literal-stop-index="59">
+                    <column-left-value name="order_id" start-index="41" stop-index="48" literal-start-index="41" literal-stop-index="48" />
+                    <operator type="=" />
+                    <column-right-value start-index="52" stop-index="59" name="order_id" />
+                </predicate>
+                <predicate start-index="65" stop-index="79" literal-start-index="65" literal-stop-index="79">
+                    <column-left-value name="order_id" start-index="65" stop-index="72" literal-start-index="65" literal-stop-index="72" />
+                    <operator type="=" />
+                    <compare-right-value>
+                        <literal-expression value="1000" start-index="76" stop-index="79" />
+                    </compare-right-value>
+                </predicate>
+            </and-predicate>
+        </where>
+    </update>
 </sql-parser-test-cases>

